### ISSUE TITLE
[Bug][Grid] GridColumnMenuFilter items cannot be reached

### ISSUE THREAD

**Issue by filipKovachev** [telerik/kendo-react#3409]
### Description

Currently when opening the `GridColumnMenuFilter`, closing it and then navigating back, the filter items cannot be reached via keyboard.

### Plan

KendoReact

### Steps To Reproduce

1. Open this example: https://stackblitz.com/edit/react-pf8bh3t3?file=app%2Fapp.tsx
2. Press the Tab key until the focus lands on the column menu icon. Press the Enter key to expand the column menu pop-up
3. Navigate to the Filter section by pressing the Tab key and pressing the Enter key
4. Once the Filter expands, navigate through the rest of the filter menu items until the focus returns to the first item in the column menu popup.
5. Navigate back to the Filter section, and continue pressing the Tab key. The filter menu items can't be reached by keyboard anymore

### Screenshots or video

The described behavior can be seen in action here: https://jam.dev/c/72519025-5b6d-40ce-b87e-16c832bebec7

### Actual Behavior

When navigating the Grid column menu using the keyboard, focus cannot return to the Filter section after it has been expanded once. Once the user cycles through the menu and tries to navigate back, the filter menu items are no longer reachable via keyboard. As a result, keyboard-only users cannot access or modify filter options without using the mouse.

### Expected Behavior

Users should be able to fully navigate to and within the Filter section of the Grid column menu using the keyboard at any time. After expanding or collapsing the Filter menu, all filter options should remain reachable and operable through keyboard navigation without requiring mouse interaction

### Browser

Chrome

### Browser version

latest

### OS type

MacOS

Ticket ID: 1701243

**Comment by vveesseelliinnaa** [comment]
Still reproducible in v 13.3.0 - https://stackblitz.com/edit/react-pf8bh3t3-re3h2zph?file=package.json

**Comment by Sasho32** [comment]
The fix is available in v14.5.0-develop.16.
